src/OpenApi/ClientGenerator.php

```diff
@@ -388,6 +397,43 @@ protected function addBodyParam(array $arguments, array &$code): void
         }
     }
 
+    protected function addFormData(array $arguments, array &$code): void
+    {
+        $formData = [];
+        $multipart = [];
+
+        foreach ($arguments as $argument) {
+            if ($argument['in'] != 'formData') {
+                continue;
+            }
+
+            if ($argument['is_file']) {
+                $multipart[] = '                [';
+                $multipart[] = sprintf('                    \'name\' => \'%s\',', $argument['name']);
+                $multipart[] = sprintf('                    \'contents\' => fopen($%s, \'r\'),', $argument['name']);
+                $multipart[] = '                ],';
+            } else {
+                $formData[] = sprintf('                \'%s\' => $%s,', $argument['name'], $argument['name']);
+            }
+        }
+
+        if ($formData) {
+            $code[] = '            \'form_params\' => [';
+            foreach ($formData as $argument) {
+                $code[] = $argument;
+            }
+            $code[] = '            ],';
+        }
```

Comment:
I believe this is here to support both file and non-file parameters. However, this is currently not used by any method.

Also, in Guzzle 7 it's not possible to use `multipart` and `form_params` at the same time. This will trigger the following error: `You cannot use form_params and multipart at the same time. Use the form_params option if you want to send application/x-www-form-urlencoded requests, and the multipart option to send multipart/form-data requests.`

This can be solved by adding non-file parameters with `"in" : "formData"` to `query` in the `$options` array. But you could also just drop this, as `"in" : "formData"` is currenly only used in the specs for file upload.

Reply:
will look into this